modules/juce_core/native/juce_linux_SystemStats.cpp

```diff
@@ -36,17 +36,16 @@ SystemStats::OperatingSystemType SystemStats::getOperatingSystemType()
 
 String SystemStats::getOperatingSystemName()
 {
+    auto result = readPosixConfigFileValue ("/etc/lsb-release", "DISTRIB_ID");
+    if (result.isNotEmpty())
+        return result;
+
     return "Linux";
 }
 
 bool SystemStats::isOperatingSystem64Bit()
 {
-   #if JUCE_64BIT
-    return true;
-   #else
-    //xxx not sure how to find this out?..
-    return false;
-   #endif
+    return (sizeof (void*) * CHAR_BIT) == 64;

```

Comment:
When was the previous check not sufficient? Were you seeing issues on a certain device?

Reply:
I'm trying to find a less `JUCE_*` macro-y way of doing it as it would be totally specific to JUCE itself. Like I mentioned on the forum post - open to any alternative. I think this case warrants a dynamic alternative, otherwise why bother with a platform specific solution in this function?

Reply:
That sizeof() is checking at compile time. Try building a test program with `gcc -m32 foo.c` on 64bit Linux to check. In that case (compiling 32bit and running on 64bit OS) it will return 32bit. 

You may want to try `uname()` and looking at the machine string. But then you will need to make a list of possible JUCE supported `uname -m` strings and whether they are 64bit. This becomes difficult if it needs updating in future as new ARM arch are introduced for instance.

Reply:
Good point. I'll dig around to see if there's a list of example return types and get rolling on fixing this up...

Reply:
@hjmallon If I've done my homework correctly, the new commit I pushed should satisfy the majority of cases. Let me know what you think.

Reply:
I would be tempted to leave the #ifdef JUCE_64BIT, return true section in as that will always (until JUCE supports 128bit OSs) be true and that prevents any regressions here.

Reply:
Since the goal is to dynamically check this detail, I don't see what's wrong with this route that I've taken at this point. Besides, this would (at best) land in `develop` which is _the_ place to catch regressions. And I highly doubt JUCE will support a 128bit OS - they're barely useful enough to matter as it is...

> Also check the return value

Mm... why bother? Worst case here is that `machine` is null or empty, which will fail all conditions.

Reply:
I see an improvement already where the case should be ignored.